components/dashboard/src/settings/Preferences.tsx

```diff
@@ -19,70 +18,57 @@ import settingsMenu from "./settings-menu";
 
 type Theme = 'light' | 'dark' | 'system';
 
+const DesktopNoneId = "none"
+const DesktopNoneLogoDark = "image/none-dark.svg"
+const DesktopNoneLogoLight = "image/none.svg"
+const DesktopNone: IDEOption = {
+    "image": "",
+    "logo": DesktopNoneLogoLight,
+    "orderKey": "-1",
+    "title": "None",
+    "type": "desktop"
+}
+
 export default function Preferences() {
     const { user } = useContext(UserContext);
-    const { setIsDark } = useContext(ThemeContext);
-
-    const [defaultIde, setDefaultIde] = useState<string>(user?.additionalData?.ideSettings?.defaultIde || "");
-    const actuallySetDefaultIde = async (value: string) => {
-        const additionalData = user?.additionalData || {};
-        const settings = additionalData.ideSettings || {};
-        settings.defaultIde = value;
+    const { setIsDark, isDark } = useContext(ThemeContext);
+
+    const updateUserIDEInfo = async (defaultDesktopIde: string, defaultIde: string) => {
+        let useDesktopIde = defaultDesktopIde !== DesktopNoneId;
+        const additionalData = user?.additionalData ?? {};
+        const settings = additionalData.ideSettings ?? {};
+        settings.useDesktopIde = useDesktopIde;
+        settings.defaultIde = defaultIde;
+        settings.defaultDesktopIde = defaultDesktopIde;
         additionalData.ideSettings = settings;
         getGitpodService().server.trackEvent({
             event: "ide_configuration_changed",
             properties: {
                 useDesktopIde,
-                defaultIde: value,
+                defaultIde,
                 defaultDesktopIde: useDesktopIde ? defaultDesktopIde : undefined
             },
-        });
+        }).then().catch(console.error);
```

Comment:
**Question**: Why is change needed? (Doesn't the console already log uncaught Promise errors?)

Reply:
Mostly because that it will crash in Nodejs v16, so it's good to add this for `Promise`